code/game/machinery/vending.dm

```diff
@@ -1935,6 +1935,59 @@ var/global/num_vending_terminals = 1
 
 	pack = /obj/structure/vendomatpack/engivend
 
+/obj/machinery/vending/building
+	name = "\improper Habitat Depot"
+	desc = "Habitat, sweet habitat. All you need for remodeling."
+	icon_state = "building"
+	products = list(
+		/obj/item/stack/sheet/metal/bigstack = 10,
```

Comment:
I think you might be right, I'll do some testing, that would obviously be preferable if for aesthetic reasons alone. I didn't notice any stacking issues when testing, but I suspect (and never tested) that it might have some issues with recycling the proper amount of minerals.